core/org.openjdk.jmc.agent/src/main/java/org/openjdk/jmc/agent/util/TypeUtils.java

```diff
@@ -227,14 +230,71 @@ public static boolean shouldStringify(Type argumentType) {
 	/**
 	 * Transforms a FQN in internal form, so that it can be used in e.g. formal descriptors.
 	 *
-	 * @param className
-	 *            the fully qualified class name in internal form.
+	 * @param className the fully qualified class name in internal form.
 	 * @return the transformed class name.
 	 */
 	public static String parameterize(String className) {
 		return "L" + className + ";"; //$NON-NLS-1$ //$NON-NLS-2$
 	}
 
+	public static String getInternalName(String className) {
+		return className.replace('.', '/');
+	}
+
+	public static String getCanonicalName(String binaryName) {
+		return binaryName.replace('/', '.');
+	}
+
+	public static int getConstZeroOpcode(Type type) {
+		switch (type.getSort()) {
+		case Type.BOOLEAN:
+		case Type.BYTE:
+		case Type.CHAR:
+		case Type.SHORT:
+		case Type.INT:
+			return Opcodes.ICONST_0;
+		case Type.FLOAT:
+			return Opcodes.FCONST_0;
+		case Type.LONG:
+			return Opcodes.LCONST_0;
+		case Type.DOUBLE:
+			return Opcodes.DCONST_0;
+		case Type.ARRAY:
+		case Type.OBJECT:
+			return Opcodes.ACONST_NULL;
+		case Type.METHOD:
+		case Type.VOID:
+			throw new UnsupportedOperationException();
+		default:
+			throw new AssertionError();
+		}
+	}
+
+	public static Object getFrameVerificationType(Type type) {
```

Comment:
In general, javadocs for public methods.

Reply:
It has been updated with Javadoc. Please refer to the latest changeset.